From: Qi Zheng <zhengqi.arch@bytedance.com>
Subject: mm: thp: fix build error with CONFIG_SHMEM disabled
Date: Thu, 1 Sep 2022 15:44:42 +0800

Fix the following build error by changing function
parameter type from struct mm_sot * to struct khugepaged_mm_slot *:

mm/khugepaged.c: In function 'khugepaged_scan_mm_slot':
 >> mm/khugepaged.c:2056:45: error: passing argument 1 of
'khugepaged_collapse_pte_mapped_thps' from incompatible pointer type
[-Werror=incompatible-pointer-types]
     2056 |         khugepaged_collapse_pte_mapped_thps(mm_slot);
          |                                             ^~~~~~~
          |                                             |
          |                                             struct
khugepaged_mm_slot *
    mm/khugepaged.c:2023:65: note: expected 'struct mm_slot *' but
argument is of type 'struct khugepaged_mm_slot *'
     2023 | static void khugepaged_collapse_pte_mapped_thps(struct
mm_slot *mm_slot)
          |
~~~~~~~~~~~~~~~~^~~~~~~
    cc1: some warnings being treated as errors

Link: https://lkml.kernel.org/r/639fa8d5-8e5b-2333-69dc-40ed46219364@bytedance.com
Signed-off-by: Qi Zheng <zhengqi.arch@bytedance.com>
Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/khugepaged.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/khugepaged.c~mm-thp-convert-to-use-common-struct-mm_slot-fix
+++ a/mm/khugepaged.c
@@ -2017,7 +2017,7 @@ static int khugepaged_scan_file(struct m
 	BUILD_BUG();
 }
 
-static void khugepaged_collapse_pte_mapped_thps(struct mm_slot *mm_slot)
+static void khugepaged_collapse_pte_mapped_thps(struct khugepaged_mm_slot *mm_slot)
 {
 }
 #endif
_
